### PR TITLE
Add Vlan and MTU to switch_macmap output

### PR DESCRIPTION
**Pull request by cxhong**
for issue #3305 .  the output of switch_macmap will look (as example from our BNT switch) :
````
[root@fs3 xCAT_plugin]# xcatprobe switch_macmap mgmtsw01
Switch    Port        MAC address         Node                  Vlan     MTU
--------------------------------------------------------------------------------------
mgmtsw01  Ethernet1   70:e2:84:14:09:f6   sn01                  553      9216
mgmtsw01  Ethernet1   70:e2:84:14:09:f5   sn01                  553
mgmtsw01  Ethernet10  8c:ea:1b:e8:78:c0   mid05tor10            553      9216
mgmtsw01  Ethernet11         N/A                                         9216
mgmtsw01  Ethernet12  8c:ea:1b:12:ca:40   mid05tor12            553      9216
mgmtsw01  Ethernet13         N/A                                         9216
mgmtsw01  Ethernet14         N/A                                         9216
mgmtsw01  Ethernet15         N/A                                         9216
mgmtsw01  Ethernet16         N/A                                         9216
mgmtsw01  Ethernet17         N/A                                         9216
mgmtsw01  Ethernet18         N/A                                         9216
mgmtsw01  Ethernet19         N/A                                         9216
mgmtsw01  Ethernet2   70:e2:84:14:0a:70   sn02                  553      9216
mgmtsw01  Ethernet2   70:e2:84:14:0a:71   sn02                  553
mgmtsw01  Ethernet20         N/A                                         9216
mgmtsw01  Ethernet21         N/A                                         9216
mgmtsw01  Ethernet22  0c:c4:7a:d2:cf:f8                         1        9216
mgmtsw01  Ethernet23         N/A                                         9216
................................
mgmtsw01  Ethernet48  6c:ae:8b:1b:e8:56                         105
mgmtsw01  Ethernet48  42:fb:0a:05:69:c9                         105
mgmtsw01  Ethernet48  34:40:b5:b9:8b:5a                         105
mgmtsw01  Ethernet49         N/A                                         9216
mgmtsw01  Ethernet5   f4:52:14:d6:bb:10   mid05                 553      9216
mgmtsw01  Ethernet50         N/A                                         9216
mgmtsw01  Ethernet51         N/A                                         9216
mgmtsw01  Ethernet52  7c:fe:90:f2:21:42                         553      9216
mgmtsw01  Ethernet52  70:e2:84:14:0f:08                         553
mgmtsw01  Ethernet52  00:02:c9:58:d3:d9                         553
mgmtsw01  Ethernet6          N/A                                         9216
mgmtsw01  Ethernet7          N/A                                         9216
mgmtsw01  Ethernet8          N/A                                         9216
mgmtsw01  Ethernet9          N/A                                         9216
--------------------------------------------------------------------------------------
```

### CODE REVIEW

File: perl-xCAT/xCAT/MacMap.pm [hunk 1]
```diff
@@ -783,24 +791,38 @@ sub refresh_switch {
...
             my %index_to_mac = ();
+            my %index_to_vlan = ();
             foreach (keys %$mactoindexmap) {
                 my $index     = $mactoindexmap->{$_};
                 my @tmp       = split /\./, $_;
+                my $vlan      = @tmp[0];
                 my @mac       = @tmp[ -6 .. -1 ];
                 my $macstring = sprintf("%02x:%02x:%02x:%02x:%02x:%02x", @mac);
                 push @{ $index_to_mac{$index} }, $macstring;
+                push @{ $index_to_vlan{$index} }, $vlan;
```
Comment (zet809) [inline perl-xCAT/xCAT/MacMap.pm]: Hi, @cxhong , the vlan can be tagged on either swithport or the host NIC port. 
In this case, the VLAN is got from the package passed through a switch port, it must be tagged with the host NIC port. So, why not append the vlan id to the corresponding MAC address?

Comment (zet809) [inline perl-xCAT/xCAT/MacMap.pm]: And for MTU, it is port based, so, I think we can modify the output a little like below, what is your idea?
```
[root@fs3 xCAT_plugin]# xcatprobe switch_macmap mgmtsw01
Switch         Port(MTU)              MAC address(Vlan)         Node
-------------------------------------------------------------
mgmtsw01  Ethernet1(9216)    70:e2:84:14:09:f6(553)   sn01
mgmtsw01  Ethernet1(9216)    70:e2:84:14:09:f5(553)   sn01
mgmtsw01  Ethernet10(9216)  8c:ea:1b:e8:78:c0(553)   mid05tor10
```

**Comment by whowutwut** [comment]
@zet809 So the idea here is to create a way to more easily verify that the switch configuration is correct when VLAN comes into play.  So I think we need to think a little about the use case and how it would help us determine whether switches and uplink switches are configured correctly when we are unable to communicate to the compute nodes at the leaf switches..

File: perl-xCAT/xCAT/MacMap.pm (hunk 1 above)
Comment (cxhong) [inline perl-xCAT/xCAT/MacMap.pm]: Thanks, @zet809 , your output looks more clear

**Comment by cxhong** [comment]
outputs:
```
[root@fs3 xCAT_plugin]# xcatprobe switch_macmap mgmtsw01
Switch    Port(MTU)           MAC address(VLAN)           Node
--------------------------------------------------------------------------------------
mgmtsw01  Ethernet1(9216)     70:e2:84:14:09:f6(553)      sn01
mgmtsw01  Ethernet1           70:e2:84:14:09:f5(553)      sn01
mgmtsw01  Ethernet10(9216)    8c:ea:1b:e8:78:c0(553)      mid05tor10
mgmtsw01  Ethernet11(9216)    N/A
mgmtsw01  Ethernet12(9216)    8c:ea:1b:12:ca:40(553)      mid05tor12
mgmtsw01  Ethernet13(9216)    N/A
mgmtsw01  Ethernet14(9216)    N/A
```

**Comment by zet809** [comment]
Hi, @cxhong , I suggest to append MTU for every switch port. Then, when we try to use 'grep' to search a specified MAC address or a switch port, we can get the full information.
```
mgmtsw01  Ethernet1(MTU size)           70:e2:84:14:09:f5(553)      sn01
```

**Comment by zet809** [comment]
Looks good to me, I agree to merge, Thx a lot @cxhong .

**Pull request merged**